src/invoice/main.mo

```diff
@@ -442,6 +460,22 @@ actor Invoice {
       canisterId = ?Principal.fromActor(Invoice);
     });
   };
+  
+  /**
+   * Adds a principal to the list of principals that can create an invoice.
+   *
+   * @param {Principal} principal
+   * @returns {()}
+   */
+  public func authorize_creation (principal: Principal) : () {
```

Comment:
Unless I'm missing something, it looks like anyone can call this method and authorize anyone else